primitive-types/src/lib.rs

```diff
@@ -110,6 +110,44 @@ impl U256 {
 	pub fn full_mul(self, other: U256) -> U512 {
 		U512(uint_full_mul_reg!(U256, 4, self, other))
 	}
+
+	/// Lossy saturating conversion from a `f64` to a `U256`.
+	///
+	/// The conversion follows roughly the same rules as converting `f64` to other
+	/// primitive integer types. Namely, the conversion of `value: f64` behaves as
+	/// follows:
+	/// - `NaN` => `0`
+	/// - `(-∞, 0]` => `0`
+	/// - `(0, u256::MAX]` => `value as u256`
+	/// - `(u256::MAX, +∞)` => `u256::MAX`
+	pub fn from_f64_lossy(value: f64) -> U256 {
+		if value >= 1.0 {
+			let bits = value.to_bits();
+			// NOTE: Don't consider the sign or check that the subtraction will
+			//   underflow since we already checked that the value is greater
+			//   than 1.0.
+			let exponent = ((bits >> 52) & 0x7ff) - 1023;
+			let mantissa = (bits & 0x0f_ffff_ffff_ffff) | 0x10_0000_0000_0000;
```

Comment:
could you please confirm it works on big-endian architechtures as well?

Reply:
Yes, this is independent of endianness since we are not indexing bytes here. That is, this is a purely uint implementation since above we see that that the bits obtained from `to_bytes` is a u64.

Reply:
> Yes, this is independent of endianness since we are not indexing bytes here. That is, this is a purely uint implementation since above we see that that the bits obtained from `to_bytes` is a u64.

So the bytes within this `u64` can be either in little-endian form or big-endian depending on our CPU.
My understanding is that `bits & 0x0f_ffff_ffff_ffff` assumes the little-endianness, please convince me that I'm wrong and this is specified by IEEE-754.
From https://en.wikipedia.org/wiki/Double-precision_floating-point_format#Endianness
> However, on modern standard computers (i.e., implementing IEEE 754), one may in practice safely assume that the endianness is the same for floating-point numbers as for integers

To be on the safer side, I'd suggest using `f64::to_be_bytes`.

Reply:
According to the docs https://doc.rust-lang.org/stable/std/primitive.f64.html#method.from_bits

> It turns out this is incredibly portable, for two reasons:
> 
> - Floats and Ints have the same endianness on all supported platforms.
> - IEEE-754 very precisely specifies the bit layout of floats.

Reply:
Ok, I looked at the generated code for a big-endian CPU and it seems to work https://godbolt.org/z/jP6rzK.
Seems like this assumption was wrong:
> My understanding is that bits & 0x0f_ffff_ffff_ffff assumes the little-endianness

Thanks for bearing with me.